### PR TITLE
[WIP] Remove Travis and Azure files

### LINKED ISSUE #6369: Include deploy and coverage in GitHub actions
Follow up to https://github.com/pytest-dev/pytest/pull/6355:

- [x] Coverage
- [x] GitHub release notes
- [x] Deploy
- [x] Cleanup Azure and Travis files

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #6369

### CODE REVIEW

**Comment by nicoddemus** [comment]
> Misses coverage then.

Yep realized that now that we are only building with GH actions. Last commit tries to improve this, let's see.

**Comment by nicoddemus** [comment]
Unfortunately had to re-add the no cover pragmas for < 3.5.1, as that Python version is not available in GitHub actions:

```
##[error]Version 3.5.1 with arch x64 not found
Available versions:

2.7.16 (x64)
3.5.7 (x64)
3.6.9 (x64)
3.7.5 (x64)
3.8.0 (x64)
```

**Comment by RonnyPfannschmidt** [comment]
how reasonable is it to make  3.5.7 the minimal supported version in the next feature release?

**Comment by nicoddemus** [comment]
> how reasonable is it to make 3.5.7 the minimal supported version in the next feature release?

No idea how widespread is Python `<3.5.2`... @The-Compiler might chime in here (he seems to be aware about this topic in general 😁 )

**Comment by blueyed** [comment]
> Unfortunately had to re-add the no cover pragmas for < 3.5.2, as that Python version is not available in GitHub actions:

Why not leave it "uncovered" then?

I'd be ok with dropping < 3.5.2, but would have like to had known it before #6435 then.. ;)  (IIRC it was said there to keep it).

**Comment by nicoddemus** [comment]
> I'd be ok with dropping < 3.5.2, but would have like to had known it before #6435 then.. ;) (IIRC it was said there to keep it).

It is not a done deal, IIRC someone had asked to keep supporting < 3.5.2... I just seem to recall @njsmith had mentioned `trio` still supported <3.5.2 so we decided to do it as well (I might be wrong though).

**Comment by The-Compiler** [comment]
> No idea how widespread is Python `<3.5.2`... @The-Compiler might chime in here (he seems to be aware about this topic in general  )

I'd say it's not very common. I dropped support for it in qutebrowser and nobody seemed to care. I've also seen various projects (e.g. [jaraco.functools](https://github.com/jaraco/jaraco.functools/blob/master/CHANGES.rst#v300)) drop 3.5 support altogether already.

Ubuntu 16.04 LTS also ships 3.5.2 (its `python3` package is at 3.5.1, but `python3.5` is at 3.5.2, and it does seem to be 3.5.2).

https://github.com/python-trio/trio/issues/880 also has some notes.

Then again, a test runner is a bit in a special position there...

**Comment by nicoddemus** [comment]
@blueyed can you please give some insights on the missing coverage? Most seems to be debug related, but I'm not seeing any configuration in Travis to suggest that those lines should be covered in the first place. Thanks!

**Comment by hugovk** [comment]
GitHub Actions:

> We'll keep Python 2.7 on the image for ~6 months after it EOL's so people have time to move over.

* https://github.com/actions/virtual-environments/issues/144#issuecomment-565045255

Which aligns quite well with "mid-2020" for pytest 4.6 supporting 2.7:

* https://docs.pytest.org/en/latest/py27-py34-deprecation.html#maintenance-of-4-6-x-versions

**Comment by blueyed** [comment]
> @blueyed a little help?

Plan to look into it later, but might take some days.

**Comment by nicoddemus** [comment]
> Plan to look into it later, but might take some days.

OK, thanks.

File: tox.ini [hunk 1]
```diff
@@ -167,7 +166,7 @@ filterwarnings =
     default:Using or importing the ABCs:DeprecationWarning:unittest2.*
     default:the imp module is deprecated in favour of importlib:DeprecationWarning:nose.*
     ignore:Module already imported so cannot be rewritten:pytest.PytestWarning
-    # produced by python3.6/site.py itself (3.6.7 on Travis, could not trigger it with 3.6.8).
+    # produced by python3.6/site.py itself
```
Comment (blueyed) [inline tox.ini]: This info could/should stay.

Comment (nicoddemus) [reply tox.ini]: Agreed. 👍

**Comment by nicoddemus** [comment]
Superseded by #6463.

**Pull request closed without merging**